ec/resources/rulesengine/v4policy.go

```diff
@@ -71,6 +71,18 @@ func ResourcePolicyCreate(
 	policyMap := make(map[string]interface{})
 	json.Unmarshal([]byte(policy), &policyMap)
 	policyMap["state"] = "locked"
+	if policyMap["name"] != nil {
```

Comment:
If a name is not provided, this value may be nil - it is possible it is an empty string. I think this needs a check as in: https://gobyexample.com/maps
`if val, ok := policyMap["name"]; ok {}` then you can cast `val` to `string` and check if `len(...) == 0`

Just to be extra safe...

Reply:
I misunderstood the code, marking this resolved!